components/script/dom/webglrenderingcontext.rs

```diff
@@ -705,13 +705,16 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
             return self.webgl_error(InvalidEnum);
         }
 
-        if let Some(renderbuffer) = renderbuffer {
-            renderbuffer.bind(target)
-        } else {
-            // Unbind the currently bound renderbuffer
-            self.ipc_renderer
-                .send(CanvasMsg::WebGL(WebGLCommand::BindRenderbuffer(target, None)))
-                .unwrap()
+        match renderbuffer {
+            Some(renderbuffer) if !renderbuffer.is_deleted() => {
```

Comment:
Good catch! I think this behavior (unbinding even if the renderbuffer is deleted) is tricky. Could you clarify this? A quote to the GLES 2.0 spec like:

> A renderbuffer object binding created with glBindRenderbuffer remains active **until a different renderbuffer object name is bound**, or until the bound renderbuffer object is deleted with glDeleteRenderbuffers.

(emphasis mine).

Would be helpful, assuming that *until a different renderbuffer object is bound* catches the case of naming a deleted renderbuffer? But it seems to me that after calling DeleteRenderbuffer, we shouldn't consider that a valid renderbuffer, so we should silently ignore it, instead of unbinding the current one. This would be quoting the [GLES 2.0 glDeleteRenderbuffers page](https://www.khronos.org/opengles/sdk/docs/man/xhtml/glDeleteRenderbuffers.xml):

> After a renderbuffer object is deleted, it has no contents, and its name is free for reuse[...]

And the WebGL 1.0 spec, that only specifies:

>  If renderbuffer is null the renderbuffer object currently bound to this target is unbound.

In fact, just checked the mesa src, and what they seem to do in `glDeleteRenderbuffers`, is deleting immediately the renderbuffer from the hash table, so when the name is given to `glBindRenderbuffer` it's not found.

In the not found case (and when `allow_user_names` is `false`, that I guess it's on core profile, they generate an `INVALID_OPERATION` error.

So given the `INVALID_OPERATION` error is not spec'd in WebGL nor GLES2, nor unbinding the current renderbuffer, I think the sensitive thing is ignoring it. What do you think?